processors/blockprocessor/block_processor.go

```diff
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/algorand/indexer/util/metrics"
```

Comment:
Can you register this without adding a common dependency?

Two possible options:
1) similar to `OnComplete(rnd) error`, a dynamic `ProvideMetrics() []prometheus.Collector` function could be used to fetch the metrics during init.
2) we could say the global metrics are always used and call `prometheus.Register` in the init function.

Reply:
implemented option 1.